src/components/grid-aware/Modal/Modal.tsx

```diff
@@ -1,29 +1,38 @@
-import PropTypes from "prop-types";
-import React from "react";
+import * as React from "react";
 import ReactModal from "react-modal";
 
 import s from "./Modal.module.css";
 import closeIcon from "./close-icon.svg";
 
+type ModalProps = {
+  children: React.ReactNode;
+  isOpen: boolean;
+  setIsOpen: (x: boolean) => null;
```

Comment:
You should actually specify a return type of [`void`](https://www.typescriptlang.org/docs/handbook/2/functions.html#void) here. The difference between `void` and `null` is that `null` is a real JavaScript value that can be returned, while `void` is a TypeScript only type indicates that a function has no real return value. With a function that returns `null`, it is legal to assign that return value to a variable that accepts `null` as a potential value, whereas with a function that has a `void` return type, it is illegal to try to do anything with the return value.